### PR TITLE
Implement Oracle UCP connection pool metrics

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        .waitAndAssertMetrics(
+            INSTRUMENTATION_NAME, "db.client.connections.usage", AbstractIterableAssert::isEmpty);
+    testing()
+        .waitAndAssertMetrics(
+            INSTRUMENTATION_NAME, "db.client.connections.max", AbstractIterableAssert::isEmpty);
+    testing()
+        .waitAndAssertMetrics(
+            INSTRUMENTATION_NAME,
+            "db.client.connections.pending_requests",
+            AbstractIterableAssert::isEmpty);
```
Comment (trask) [inline instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java:92]: same question from https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6102/files#r883947307

> is there a need for waiting here vs a single assert?

**Review by trask — APPROVED** [review]

File: instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java (hunk 1 above)
Comment (laurit) [inline instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java:92]:
```suggestion
    Set<String> metricNames =
        new HashSet<>(
            Arrays.asList(
                "db.client.connections.usage",
                "db.client.connections.max",
                "db.client.connections.pending_requests"));
    assertThat(testing().metrics())
        .filteredOn(
            metricData ->
                metricData.getInstrumentationScopeInfo().getName().equals(INSTRUMENTATION_NAME)
                    && metricNames.contains(metricData.getName()))
        .isEmpty();
```
@trask is this what you meant?

Comment (trask) [reply instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java:92]: ya, though maybe we also need to sleep (at least time for metric interval) before asserting

Comment (breedx-splk) [reply instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java:92]: > ya, though maybe we also need to sleep (at least time for metric interval) before asserting

Hmmm, do you think we're just getting lucky now then?

Comment (laurit) [reply instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java:92]: There is a sleep right before the snippet starts and you are correct, it is needed.

Comment (trask) [reply instrumentation/oracle-ucp-11.2/testing/src/main/java/io/opentelemetry/instrumentation/oracleucp/AbstractOracleUcpInstrumentationTest.java]: oh, indeed, thx!

**Pull request merged**